client/my-sites/email/gsuite-add-users/index.jsx

```diff
@@ -34,8 +46,77 @@ import SectionHeader from 'components/section-header';
 import QueryEmailForwards from 'components/data/query-email-forwards';
 import QueryGSuiteUsers from 'components/data/query-gsuite-users';
 import getGSuiteUsers from 'state/selectors/get-gsuite-users';
+import { recordTracksEvent as recordTracksEventAction } from 'state/analytics/actions';
+
+/**
+ * Style dependencies
+ */
+import './style.scss';
 
 class GSuiteAddUsers extends React.Component {
+	state = {
+		users: [],
+	};
+
+	static getDerivedStateFromProps(
+		{ domains, isRequestingDomains, selectedDomainName },
+		{ users }
+	) {
+		if ( ! isRequestingDomains && 0 === users.length ) {
+			const domainName = getEligibleGSuiteDomain( selectedDomainName, domains );
+			if ( '' !== domainName ) {
+				return {
+					users: newUsers( domainName ),
+				};
+			}
+		}
+		return null;
+	}
+
+	onUsersChange = users => {
+		this.setState( {
+			users,
+		} );
+	};
+
+	recordContinueEvent = () => {
+		const { recordTracksEvent, selectedDomainName } = this.props;
+		const { users } = this.state;
+		recordTracksEvent( 'calypso_email_management_gsuite_add_users_continue_button_click', {
+			domain_name: selectedDomainName,
+			user_count: users.length,
+		} );
+	};
+
+	recordCancelEvent = () => {
+		const { recordTracksEvent, selectedDomainName } = this.props;
+		const { users } = this.state;
+		recordTracksEvent( 'calypso_email_management_gsuite_add_users_cancel_button_click', {
+			domain_name: selectedDomainName,
+			user_count: users.length,
+		} );
+	};
```

Comment:
It looks like we could use a single method instead of those two that simply accepts an event name:

```javascript
recordEvent = ( eventName ) => {
  const { recordTracksEvent, selectedDomainName } = this.props;
  const { users } = this.state;

  recordTracksEvent( eventName, {
    domain_name: selectedDomainName,
    user_count: users.length,
  } );
};
```

This would allow us to get rid of duplication.